pkg/actions/templates.go

```diff
@@ -68,8 +68,15 @@ func AddTemplateRepo(c *cli.Context) {
 	url := c.String("url")
 	desc := c.String("description")
 	name := c.String("name")
+	username := c.String("username")
+	password := c.String("password")
+
+	gitCredentials := utils.GitCredentials{
+		Username: username,
+		Password: password,
```

Comment:
Do we need to check if the user has provided both rather than just one or the other? Or do we rely on portal to do that? We do check the URL in `AddTemplateRepo()`. We don't check description but that's optional and doesn't depend on another parameter.

Reply:
I can add that check in a follow up PR, when we allow the user to authenticate via an access token

Curretnly:
`cwctl templates repos add` just passes whatever username and password it gets to PFE, without checking first what it's received. PFE master does not currently verify that it received both username and password. But my PFE Access Token PR does check this (https://github.com/eclipse/codewind/pull/2762)

Reply:
It makes sense to do this in the follow up when we have another authentication field to pass that you wouldn't use with the other one. I'll approve and merge this PR.

Reply:
thanks, I've implemented your suggestion in https://github.com/eclipse/codewind-installer/pull/453